### PR TITLE
Replace {YYYY} in the copyright line by the current year

### LINKED ISSUE #36776: [Issue] Replace {YYYY} in the copyright line by the current year
This issue is automatically created based on existing pull request: magento/magento2#36754: Replace {YYYY} in the copyright line by the current year

---------
Follow up for #5004

### Description (*)

So many people spend time updating the copyright in their Magento installations every year. We want to save this time by introducing a simple solution - add a variable that will be automatically replaced with a current year.

### Related Pull Requests
https://github.com/magento/magento2/pull/5004

### Manual testing scenarios (*)
1. Set copyright string to Copyright 2013-{YYYY} in the design config
2. Clear caches
3. Check the frontend
4. It displays (in this year) Copyright 2013-2023

### Questions or comments

**A note about caching**: The currrent year is evaluated when the page is rendered. As pages might be in the full page cache, the year will not immediately update in midnight Jan 01st 0:00. Mitigating this would cause a lot of problems and complicate the solution. For example we might not want to clear the full page cache automatically as this will slow down the page.

When a page is invalidated by other reasons (for example a product update), the copyright year will update as well.
So in the end of the day, the copyright year is current when a page was recently changed which is just what a copyright year is supposed to be. If a page is not updated for a long time, it's believed to be okay to have an older copyright year.

We also added a note to hint users who want a consistent up-to-date copyright year to clear the full cache.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by amenk**
Follow up for #5004

### Description (*)

So many people spend time updating the copyright in their Magento installations every year. We want to save this time by introducing a simple solution - add a variable that will be automatically replaced with a current year.

### Related Pull Requests
https://github.com/magento/magento2/pull/5004

### Manual testing scenarios (*)
1. Go to Cotent >> Design >> Configuration >> Edit (Global)>> Footer and Set copyright string to Copyright 2013-{YYYY} in the design config 
2. Clear caches
3. Check the frontend
4. It displays (in this year) Copyright 2013-2023

### Questions or comments

**A note about caching**: The currrent year is evaluated when the page is rendered. As pages might be in the full page cache, the year will not immediately update in midnight Jan 01st 0:00. Mitigating this would cause a lot of problems and complicate the solution. For example we might not want to clear the full page cache automatically as this will slow down the page.

When a page is invalidated by other reasons (for example a product update), the copyright year will update as well.
So in the end of the day, the copyright year is current when a page was recently changed which is just what a copyright year is supposed to be. If a page is not updated for a long time, it's believed to be okay to have an older copyright year.

We also added a note to hint users who want a consistent up-to-date copyright year to clear the full cache.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#36776: Replace {YYYY} in the copyright line by the current year

### CODE REVIEW

**Comment by amenk** [comment]
@magento run all tests

File: app/code/Magento/Theme/Block/Html/Footer.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ public function getCopyright()
                 \Magento\Store\Model\ScopeInterface::SCOPE_STORE
             );
         }
-        return __($this->_copyright);
+        return $this->replaceCurrentYear(__($this->_copyright));
```
Comment (ihor-sviziev) [inline app/code/Magento/Theme/Block/Html/Footer.php]:
```suggestion
        return $this->replaceCurrentYear((string)__($this->_copyright));
```

File: app/code/Magento/Theme/Block/Html/Footer.php [hunk 2]
```diff
@@ -133,4 +133,12 @@ protected function getCacheLifetime()
     {
         return parent::getCacheLifetime() ?: 3600;
     }
+    
+    /**
+     * Replace YYYY by the current year
+     */
+    private function replaceCurrentYear($text)
```
Comment (ihor-sviziev) [inline app/code/Magento/Theme/Block/Html/Footer.php]:
```suggestion
    private function replaceCurrentYear(string $text): string
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@amenk, could you please cover this change with a unit test?

**Comment by amenk** [comment]
Okay, need to find some time.
Also we should at a note on the config entry in the admin panel for the Copyright line to document this feature.

**Comment by amenk** [comment]
@magento run all tests

File: app/code/Magento/Theme/Block/Html/Footer.php [hunk 3]
```diff
@@ -133,4 +133,12 @@ protected function getCacheLifetime()
     {
         return parent::getCacheLifetime() ?: 3600;
     }
+
+    /**
+     * Replace YYYY by the current year
```
Comment (ihor-sviziev) [inline app/code/Magento/Theme/Block/Html/Footer.php]:
```suggestion
     * Replace YYYY with the current year
     *
     * @param string $text
```

File: app/code/Magento/Theme/Test/Unit/Block/Html/FooterTest.php [hunk 4]
```diff
@@ -23,14 +28,41 @@ class FooterTest extends TestCase
     protected function setUp(): void
```
Comment (ihor-sviziev) [inline app/code/Magento/Theme/Test/Unit/Block/Html/FooterTest.php:33]:
```suggestion
    /**
     * @var Config
     */
    private $scopeConfig;
    
    protected function setUp(): void
```

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Lima** [comment]
@magento create issue

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento `2.4-develop`

**Manual testing scenario:**
1. Go to Content >> Design >> Configuration >> Edit (Global)>> Footer and Set copyright string to Copyright 2013-{YYYY} in the design config
2. Clear caches
3. Check the frontend footer.

**Before:** :heavy_multiplication_x: Current year was not displaying in the footer of frontend.
Screenshot 2023-01-25 at 11 11 44 AM


**After:** :heavy_check_mark:  Now displaying after taking PR changes.
Screenshot 2023-01-25 at 11 19 12 AM

**Additionally tested:**

- Changed the theme and verified the footer.
- Created order from guest and customer.
- Invoice,shipped and credit memo of the order from the admin.

No additional manual test case is required as part of regression.

Build is failed hence, moving this PR to extended testing.

**Comment by engcom-Echo** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Echo** [comment]
Functional Tests B2B failure are different on last two run.
Run1
image

Run2
image

Hence moving to `Merge In Progress`.

**Review by fredden — APPROVED** [review]

**Comment by engcom-Lima** [comment]
Looking like bot moved to Ready for Testing. Hence, moving back to Merge in progress.

**Pull request merged**